### PR TITLE
RavenDB-13603: v4.2 on Windows 7 / Server 2008 ("Windows Ver < 6.2")

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Sparrow.Server/Platform/Pal.cs [hunk 1]
```diff
@@ -34,7 +34,18 @@ static Pal()
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                fromFilename = Environment.Is64BitProcess ? $"{toFilename}.win.x64.dll" : $"{toFilename}.win.x86.dll";
+                var win7 = "";
+                var osdesc = RuntimeInformation.OSDescription.Split(new[] { "Windows " }, StringSplitOptions.RemoveEmptyEntries);
+                if (osdesc.Length > 1)
```
Comment (ppekrol) [inline src/Sparrow.Server/Platform/Pal.cs]: use `PlatformDetails.IsWindows8OrNewer` for this.

File: src/Raven.Pal/build-all-windows.bat [hunk 2]
```diff
@@ -13,18 +13,38 @@ cmd.exe /c ""%vcbin%\amd64\vcvars64.bat" & "%vcbin%\amd64\cl" -Felibrvnpal.x64.d
 copy librvnpal.x64.dll artifacts\librvnpal.win.x64.dll
 del *.obj librvnpal.x*
 
-echo =================================================
+cmd.exe /c ""%vcbin%\vcvars32.bat" & "%vcbin%\cl" -Felibrvnpal.win7x86.dll -I inc /D RVN_WIN7 /D WINVER=0x0601 /D _WIN32_WINNT=0x0601 /D NTDDI_VERSION=0x6010000 /D WINNT=1 /O2 /analyze /sdl /LD src\*.c src\win\*.c /link"
```
Comment (ppekrol) [inline src/Raven.Pal/build-all-windows.bat:16]: I think it would be best to add those `WINVER` (and friends) to the other DLLs. Those need to match Win8.

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Sparrow.Server/Platform/Pal.cs [hunk 3]
```diff
@@ -34,7 +34,18 @@ static Pal()
...
+                {
```
Comment (ayende) [inline src/Sparrow.Server/Platform/Pal.cs]: Is there no other way to do this?

File: src/Sparrow.Server/Platform/Pal.cs (hunk 1 above)
Comment (aviviadi) [reply src/Sparrow.Server/Platform/Pal.cs]: We said do not want to use that

Comment (ppekrol) [reply src/Sparrow.Server/Platform/Pal.cs]: Why we do not want to use that?

Comment (aviviadi) [reply src/Sparrow.Server/Platform/Pal.cs]: We decided not to use PlatformDetails (and eventually throw it away).
I can copy the methods to Pal.cs though

File: src/Raven.Pal/build-all-windows.bat (hunk 2 above)
Comment (aviviadi) [reply src/Raven.Pal/build-all-windows.bat:16]: Another dll for Win8 or do you want to add WIN8 flags also for WIN10 ?

File: src/Sparrow.Server/Platform/Pal.cs (hunk 1 above)
Comment (ppekrol) [reply src/Sparrow.Server/Platform/Pal.cs]: Let's use them and when we will be throwing them away, you can refactor.

File: src/Sparrow.Server/Platform/Pal.cs (hunk 3 above)
Comment (aviviadi) [reply src/Sparrow.Server/Platform/Pal.cs]: OSDescription can give you Windows OS version in dotnet core.. We can parse it in many ways, but I am not familiar with another way to get the OS version in non-string corefx call

File: src/Sparrow.Server/Platform/Pal.cs (hunk 1 above)
Comment (aviviadi) [reply src/Sparrow.Server/Platform/Pal.cs]: done

File: src/Raven.Pal/build-all-windows.bat (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Pal/build-all-windows.bat:16]: single dll, exactly the one that we have now, but with correct flags, so we will not add features that we do not have in the future.

**Pull request merged**